### DISCUSSION TITLE
how to use loops in duckdb? for example a foreach loop (documentation not clear in my opinion)

### DISCUSSION THREAD

**Discussion by suiluj** [duckdb/duckdb#13034] [General]
**Update: ist this loop feature only part of a testing suite?**

Hello I am trying to union all multiple sql queries base on a result of a previous query.

At the moment I am not sure how to do it.

I found the section *Loops* in the duckdb documentation: https://duckdb.org/docs/dev/sqllogictest/loops

So I wanted to try that in general. But I do not know how to start. The syntax ist not clear and I cannot follow the example.

I would like to do a query and then follow up queries based on column values of the previous one. At the end I am planning to union all of the follow up queries.

So I am guessing a `foreach` loop could be interesting.

Does someone have more/better examples for loops in duckdb?

**Comment by arjenpdevries** [comment]
Those docs describe syntax of the testing framework, not the DB itself.

**Reply by Tishj** [reply]
Yes it's just the testing framework that has this
